dbt/adapters/spark/impl.py

```diff
@@ -231,8 +231,20 @@ def get_columns_in_relation(self, relation: Relation) -> List[SparkColumn]:
             # return relation's schema. if columns are empty from cache,
             # use get_columns_in_relation spark macro
             # which would execute 'describe extended tablename' query
-            rows: List[agate.Row] = super().get_columns_in_relation(relation)
-            columns = self.parse_describe_extended(relation, rows)
+            try:
+                rows: List[agate.Row] = super().get_columns_in_relation(relation)
+                columns = self.parse_describe_extended(relation, rows)
+            except dbt.exceptions.RuntimeException as e:
+                # spark would throw error when table doesn't exist, where other
+                # CDW would just return and empty list, normalizing the behavior here
+                errmsg = getattr(e, "msg", "")
+                if (
+                    f"Table or view not found: {relation}" in errmsg or
+                    "NoSuchTableException" in errmsg
```

Comment:
This doesn't match the error message from DBR 9.1 LTS.

For an invalid query:

```sql
describe extended test16497061710125262902_test_basic.model
```
, the error message from DBR 9.1 LTS is:
```
Error in SQL statement: AnalysisException: Table or view not found for 'DESCRIBE TABLE': test16497061710125262902_test_basic.model; line 1 pos 0;
...
```

Do we have a minimum version of DBR we should support? If we support DBR 9.1 LTS, we should fix this.

Reply:
@ueshin Very open to your input re: minimum supported version of DBR! My opinion on this is not tremendously well informed, but all things being equal, I think the `dbt-spark` plugin should aim to support Spark 3.0+

Reply:
@ueshin  I updated this in #331. Let me know if anything else we should change.
@jtcohen6  we currently don't have anything running tests on Spark 3.0+ do we?

Reply:
@ChenyuLInx I confirmed the tests against DBR 9.1 LTS with the latest changes have passed. Thanks!